portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -4469,6 +4470,33 @@ public User updateEmailAddressVerified(
 		return userPersistence.update(user);
 	}
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public User updateExternalReferenceCode(
+			long userId, String externalReferenceCode)
+		throws PortalException {
+
+		return updateExternalReferenceCode(
+			getUserById(userId), externalReferenceCode);
+	}
+
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public User updateExternalReferenceCode(
+			User user, String externalReferenceCode)
+		throws PortalException {
+
+		if (Objects.equals(
```

Comment:
Probably we need to validate the ERC, it should be not empty and unique

Reply:
After sending a PATCH using a existing ERC, I received :
	
Error: response status is 400

Response body
Download
{
  "status": "BAD_REQUEST",
  "title": "This external reference code is already in use."
}

So, I don't think it's necessary